include/ccf/tx.h

```diff
@@ -171,21 +141,64 @@ namespace kv
         abstract_map, untyped_map->create_change_set(read_txid->version));
     }
 
+    std::list<AbstractHandle*> get_possible_handles(const std::string& map_name)
+    {
+      std::list<AbstractHandle*> handles;
+      auto it = all_handles.find(map_name);
+      if (it != all_handles.end())
+      {
+        for (auto& handle : it->second)
+        {
+          handles.push_back(handle.get());
+        }
+      }
+      return handles;
+    }
+
     template <class THandle>
     THandle* get_handle_by_name(const std::string& map_name)
     {
-      auto search = all_changes.find(map_name);
-      if (search != all_changes.end())
+      auto possible_handles = get_possible_handles(map_name);
```

Comment:
This refactor in `tx.h` is small steps towards it a public-only header. The problem is that while the interface we want to give users is conceptually simple (a `tx` object you can call `rw` on to get a handle, and a handle you can `get`/`put` on), the implementation is heavily templated so harder to hide. This pushes the templates up the stack slightly, so we could at this point hide most of the implementation details. But not (yet) the change set itself. I think that will be a whole PR to itself.